scipy/io/wavfile.py

```diff
@@ -728,7 +765,10 @@ def read(filename, mmap=False):
         if not hasattr(filename, 'read'):
             fid.close()
         else:
-            fid.seek(0)
```

Comment:
I am frankly mystified as to why this final seek operation was originally here. Most other libraries will not seek to the beginning of a file if you hand them an open file to read - e.g. if you use `pd.read_csv()` to read a file, the file position after the function call will be where it stopped reading.

I would suggest removing `fid.seek(0)`, and not seeking even if the file supports it.

Thoughts, @j-bowhay ?

Reply:
It was added in 47b9012, @nils-werner do you happen to remember?

Reply:
If I remember correctly it was there before, I just proposed the `if` clause. Maybe follow the `git blame` to the changes before?

I don't think it needs to be the at all tbh.